### ISSUE TITLE
Accessibility: The focus dotline rectangle for Button cannot be visible obviously when using keyboard focus it

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#4538]
* .NET Core Version: 
6.0.100-preview1.1.21107.2

* Have you experienced this same bug with .NET Framework?: 
No
* More Info: 
1. This is a regression issue, cannot repro in .Net Core 3.1 and .Net 5.0 builds
2. This issue cannot repro using mouse click buttons

**Minimal repro:**
1. Create a Winforms .Net Core app
2. Add several buttons to form designer
3. Build and run app
4. Use Tab keyboard to focus on buttons, observe

**Actual behavior:**
The focus dotline rectangle for Button cannot be visible obviously, see following screenshot and GIF for 6.0
![image](https://user-images.githubusercontent.com/26474449/107194261-c61ea580-6a2a-11eb-9743-e5396839dc9c.png)
![keyboard_Button](https://user-images.githubusercontent.com/26474449/107194327-dc2c6600-6a2a-11eb-956f-5f4c1509f913.gif)

**Expected behavior:**
The focus dotline rectangle for Button is visible obviously on .Net Core 3.1 or .Net 5.0, see following screenshot and GIF for 5.0
![image](https://user-images.githubusercontent.com/26474449/107194297-d33b9480-6a2a-11eb-98a9-ff56af3faeb3.png)
![keyboard_Button_5 0](https://user-images.githubusercontent.com/26474449/107194338-dfbfed00-6a2a-11eb-816d-e913a201729f.gif)

**Comment by wjk** [comment]
If the buttons are using the System FlatStyle, this is something we cannot fix, as the line in question is drawn by native code in user32 and uxtheme.

However, if the buttons use the Standard FlatStyle (which is the default), bitmaps of the button background are retrieved from UXTheme and drawn using standard GDI+. In this case, we can draw whatever we want on top of them. `ControlPaint.DrawFocusRectangle()` was never updated for High-DPI; it always draws a 1-physical-pixel line. This we can fix ourselves, as ControlPaint uses a managed GDI+ Pen to draw the line. I only use the System FlatStyle because then the buttons cross-fade between states like raw Win32 buttons (the Standard FlatStyle just jumps immediately between states). There is nothing to say we can't implement cross-fading on a Standard Button, as the cross-fade algorithm [is public API](https://docs.microsoft.com/en-us/windows/win32/api/uxtheme/nf-uxtheme-beginbufferedanimation). One might submit a PR adding a wrapper for BeginBufferedAnimation et al into WinForms as a standalone class (it can crossfade between any two bitmaps, not just button frames), and then rewrite Standard Button to use it.

**Comment by John-Qiao** [comment]
Verified this issue on .NET SDK 7.0.100-alpha.1.22060.5, issue was fixed.
![TestResult-4538](https://user-images.githubusercontent.com/45864985/148894407-bfe343df-907a-4096-aa42-4c47b7b39c99.gif)
